### PR TITLE
Mx image2

### PR DESCRIPTION
**Pull request by zhreshold**
unittest not available yet. A record file hosted on dmlc.data might be required before writing test cases.

### CODE REVIEW

File: python/mxnet/image/image.py [hunk 1]
```diff
@@ -112,6 +113,85 @@ def scale_down(src_size, size):
...
+            oh, ow, nh, nw = sizes
+            if nh > oh and nw > ow:
+                return 2
+            elif nh < oh and nw < ow:
+                return 3
+            else:
+                return 1
+        else:
+            return 2
+    if interp == 10:
```
Comment (piiswrong) [inline python/mxnet/image/image.py]: Why is this useful?

File: docs/api/python/io.md [hunk 2]
```diff
@@ -88,26 +89,48 @@ A list of image modification functions provided by ``mxnet.image``.
     :nosignatures:
 
     image.imdecode
+    image.get_interp_method
```
Comment (piiswrong) [inline docs/api/python/io.md]: There is enough content for a new page.
put mx.image in a new page after io.md

File: python/mxnet/image/image.py [hunk 3]
```diff
@@ -112,6 +113,85 @@ def scale_down(src_size, size):
...
+    Returns
+    -------
+    NDArray
+        Image with pixel values in range (min_val, max_val)
+    """
+    if min_val > max_val:
+        raise ValueError('min_val: %f larger than max_val: %f' % (min_val, max_val))
+    min_img = nd.full(src.shape, min_val)
+    max_img = nd.full(src.shape, max_val)
+    return nd.maximum(min_img, nd.minimum(max_img, src))
```
Comment (piiswrong) [inline python/mxnet/image/image.py]: there is nd.clip

File: python/mxnet/image/image.py [hunk 4]
```diff
@@ -308,152 +430,481 @@ def random_size_crop(src, size, min_area, ratio, interp=2):
...
+    Parameters
+    ----------
+    saturation : float
+        The saturation jitter ratio range, [0, 1]
+    min_val : float, default=0
+        Minimum value after jittering
+    max_val : float ,default=255
+        Maximum value after jittering
+    """
+    def __init__(self, saturation, min_val=0, max_val=255):
```
Comment (piiswrong) [inline python/mxnet/image/image.py]: why clip to 0, 255?
This should be applied after color normalization right?

File: python/mxnet/image/image.py (hunk 1 above)
Comment (zhreshold) [inline python/mxnet/image/image.py]: Sometimes we want random interp method in training. This is similar to GetInterMethod in image_aug_default.cc
Could be hidden function though. What do you think?

File: python/mxnet/image/image.py (hunk 4 above)
Comment (zhreshold) [inline python/mxnet/image/image.py]: @piiswrong I've noticed bad visual artifacts if we don't clip it before normalization. Clip to 0,255 make it more like a natural image.

**Comment by zhreshold** [comment]
Updated
1. moved doc to image.md
2. Use get_interp_method as internal helper function
3. Modified Colorjitter aug, saves clip operation.

**Pull request merged**